File: libraries/hlp/crehlp.f
      PROGRAM CREHLP
*+
*  - - - - - - -
*   C R E H L P
*  - - - - - - -
*
*  Create random access form of help file.
*
*
*  I/O units:
*     LUIN    input    help source
*     LUOUT   output   direct access form
*     LUCMD   input    command input
*     LUMES   output   prompts and error messages
*
*  Called:  hlp_NAMETR, hlp_CREH, hlp_ERRMES
*
*  P.T.Wallace   Starlink   13 Junly 1995
*-

      IMPLICIT NONE

*  Help library name to filename translation routine
      EXTERNAL hlp_NAMETR

*  I/O unit numbers and filenames
      INTEGER LUIN,LUOUT,LUCMD,LUMES
      CHARACTER*80 SOURCE,LIB

*  End-of-string character
      CHARACTER EOS

      INTEGER JSTAT
      CHARACTER MES*80


*-----------------------------------------------------------------------

*  Machine-dependent setup

*  I/O unit numbers
      LUIN=1
      LUOUT=2
      LUCMD=5
      LUMES=6

*  Form the "end of string" character
      EOS=CHAR(0)                         !!! ASCII-specific !!!

*  Filenames.
      SOURCE='fort.1'
      LIB='fort.2'

*-----------------------------------------------------------------------

*  Create the help file.
      CALL hlp_CREH(hlp_NAMETR,LUIN,SOURCE,LUOUT,LIB,LUMES,EOS,JSTAT)

*  Report status if bad, and exit.
      IF (JSTAT.NE.0) THEN
         CALL hlp_ERRMES(JSTAT,MES)
         WRITE (LUMES,'(1X,A)') MES
      END IF

      END

File: libraries/hlp/crehlp.c
#include <string.h>
#include "hlpsys.h"
int main ( int argc, char *argv[] )
/*
**  - - - - - - -
**   c r e h l p
**  - - - - - - -
**
**  Create random access form of HELP library file.
**
**  This is a front-end for the hlpCreh function.  It obtains the
**  names of the source and library files, and opens the files.
**
**  I/O:
**     arg1     input: HELP source
**     arg2     output: HELP library
**     stdin    command input
**     stdout   prompts and error messages
**
**  If not specified through command-line arguments, the filenames
**  are prompted for.
**
**  Returned (function value):
**               int     status: 0 = OK
**                       other -ve = error reported by hlpCreh
**
**  The error codes are defined in the hlpsys.h #include file.
**
**  Called:  hlpNametr, hlpCreh, hlpErrmes, hlpTrim
**
**  Last revision:   16 June 2000
**
**  Copyright 2000 P.T.Wallace.  All rights reserved.
*/

/* Maximum length of filenames (including '\0'). */
#define LFN 100

{
   int jstat;

/* Names of source and library files. */
   char source [ LFN ], lib [ LFN ];


/* Get the source-file name. */
   if (argc >= 2 ) {
      strncpy ( source, argv [1], LFN );
   } else {
      puts ( "Name of help source file to be read?" );
      hlpTrim ( fgets ( source, LFN, stdin ) );
   }

/* Get the library-file name. */
   if (argc >= 3 ) {
      strncpy ( lib, argv [2], LFN );
   } else {
      puts ( "Name of help library file to be written?" );
      hlpTrim ( fgets ( lib, LFN, stdin ) );
   }

/* Create the HELP library. */
   if ( ( jstat = hlpCreh ( hlpNametr, source, lib ) ) )
      puts ( hlpErrmes ( jstat ) );
   return jstat;
}
